discovery/messagestore.go

```diff
@@ -0,0 +1,204 @@
+package discovery
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+
+	"github.com/coreos/bbolt"
+	"github.com/lightningnetwork/lnd/channeldb"
+	"github.com/lightningnetwork/lnd/lnwire"
+)
+
+var (
+	// messageStoreBucket is a key used to create a top level bucket in the
+	// gossiper database, used for storing messages that are to be sent to
+	// peers. Upon restarts, these messages will be read and resent to their
+	// respective peers.
+	//
+	// maps:
+	//   pubKey (33 bytes) + msgShortChanID (8 bytes) + msgType (2 bytes) -> msg
+	messageStoreBucket = []byte("message-store")
+
+	// ErrUnsupportedMessage is an error returned when we attempt to add a
+	// message to the store that is not supported.
+	ErrUnsupportedMessage = errors.New("unsupported message type")
+
+	// ErrCorruptedMessageStore indicates that the on-disk bucketing
+	// structure has altered since the gossip message store instance was
+	// initialized.
+	ErrCorruptedMessageStore = errors.New("gossip message store has been " +
+		"corrupted")
+)
+
+// MessageStore is a store responsible for storing gossip messages which we
+// should reliably send to our peers. By design, this store will only keep the
+// latest version of a message (like in the case of multiple ChannelUpdate's)
+// for a channel with a peer.
+type MessageStore struct {
+	db *channeldb.DB
+}
+
+// NewMessageStore creates a new message store backed by a channeldb instance.
+func NewMessageStore(db *channeldb.DB) (*MessageStore, error) {
+	err := db.Update(func(tx *bbolt.Tx) error {
+		_, err := tx.CreateBucketIfNotExists(messageStoreBucket)
+		return err
+	})
+	if err != nil {
+		return nil, fmt.Errorf("unable to create required buckets: %v",
```

Comment:
nit: capitalize log msg

Reply:
Comment on incorrect line? Errors returned should be lowercase.

Reply:
oops, nope i think we're good